domain-models-runtime/src/test/java/org/folio/DemoRamlRestTest.java

```diff
@@ -170,7 +155,87 @@ private void postBook(TestContext context, String parameterString, int expectedS
+    context.assertEquals(0, books.getResultInfo().getTotalRecords());
+    context.assertEquals(0, books.getBooks().size());
+
+    buf = checkURLs(context, "http://localhost:" + port + "/rmbtests/test?query=title%3Dwater", 200);
+    books = Json.decodeValue(buf, Books.class);
+    context.assertEquals(0, books.getTotalRecords());
+
+    buf = checkURLs(context, "http://localhost:" + port + "/rmbtests/test?query=a%3D", 400);
+    context.assertTrue(buf.toString().contains("expected index or term, got EOF"));
+
+    Data d = new Data();
+    d.setAuthor("a");
+    d.setGenre("g");
+    d.setDescription("description1");
+    d.setTitle("title");
+    d.setDatetime(new Datetime());
+    d.setLink("link");
+
+    Book b = new Book();
+    b.setData(d);
+    b.setStatus(0);
+    b.setSuccess(true);
+
+    ObjectMapper om = new ObjectMapper();
+    String book = om.writerWithDefaultPrettyPrinter().writeValueAsString(b);
+
+    postData(context, "http://localhost:" + port + "/rmbtests/test", Buffer.buffer(book), 201,
+      HttpMethod.POST, "application/json", TENANT, false);
+
+    buf = checkURLs(context, "http://localhost:" + port + "/rmbtests/test", 200);
+    books = Json.decodeValue(buf, Books.class);
+    context.assertEquals(1, books.getTotalRecords());
+    context.assertEquals(0, books.getResultInfo().getDiagnostics().size());
+    context.assertEquals(1, books.getResultInfo().getTotalRecords());
+    context.assertEquals(1, books.getBooks().size());
+
+    d.setDescription("description2");
+    book = om.writerWithDefaultPrettyPrinter().writeValueAsString(b);
+    postData(context, "http://localhost:" + port + "/rmbtests/test", Buffer.buffer(book), 201,
+      HttpMethod.POST, "application/json", TENANT, false);
+
+    buf = checkURLs(context, "http://localhost:" + port + "/rmbtests/test", 200);
+    books = Json.decodeValue(buf, Books.class);
+    context.assertEquals(2, books.getTotalRecords());
+    context.assertEquals(0, books.getResultInfo().getDiagnostics().size());
+    context.assertEquals(2, books.getResultInfo().getTotalRecords());
+    context.assertEquals(2, books.getBooks().size());
+
+    // need at least one record in result before we can trigger this error
+    buf = checkURLs(context, "http://localhost:" + port + "/rmbtests/test?query=badclass%3Dtrue", 200);
+    books = Json.decodeValue(buf, Books.class);
+    context.assertEquals(2, books.getTotalRecords());
+    context.assertEquals(1, books.getResultInfo().getDiagnostics().size());
+    context.assertTrue(books.getResultInfo().getDiagnostics().get(0).getMessage().contains("Cannot deserialize instance of"));
+    context.assertEquals(2, books.getResultInfo().getTotalRecords());
+    context.assertEquals(0, books.getBooks().size());
+
+    buf = checkURLs(context, "http://localhost:" + port + "/rmbtests/test?query=slim%3Dtrue", 200);
+    JsonObject jo = new JsonObject(buf);
+    log.info("jo={}", jo.encodePrettily());
```

Comment:
Can you remove this logging or change it to debug level?

Reply:
sure